Fixes after reviewing composable player and recorder PR

MichaelOrlov opened the pull request:
- Relates https://github.com/ros2/rosbag2/pull/1419

morlov-apexai committed:
Get rid of temporal conversion crutches using map<string, Rosbag2QoS>

Signed-off-by: Michael Orlov <michael.orlov@apex.ai>

morlov-apexai committed:
Use const ref to node in the options getter functions

- Also made a style clean up in getter functions

Signed-off-by: Michael Orlov <michael.orlov@apex.ai>

morlov-apexai committed:
Cleanups in get_storage(/play/record)_options functions

Signed-off-by: Michael Orlov <michael.orlov@apex.ai>

morlov-apexai committed:
Move RosBag2RecordTestFixture insight test_record_params.cpp

Signed-off-by: Michael Orlov <michael.orlov@apex.ai>

morlov-apexai committed:
Rename overrides.yaml to the qos_profile_overrides.yaml file

Signed-off-by: Michael Orlov <michael.orlov@apex.ai>

morlov-apexai committed:
Rename params_player.yaml to the player_node_params.yaml

Signed-off-by: Michael Orlov <michael.orlov@apex.ai>

morlov-apexai committed:
Rename params_recorder.yaml to the recorder_node_params.yaml

Signed-off-by: Michael Orlov <michael.orlov@apex.ai>

morlov-apexai committed:
Replace Rosbag2Duration by rclcpp::Duration

Signed-off-by: Michael Orlov <michael.orlov@apex.ai>

MichaelOrlov commented:
cc: @roncapat Please note that this PR is still WIP. But I hope that all fixes will be eventually merged to your branch in  https://github.com/ros2/rosbag2/pull/1419

roncapat commented:
Nice! Thank you, will give a look at it RN :)

roncapat commented:
As soon as you find this PR good to merge let me know :) I gave a look and 100% agree with everything.

morlov-apexai committed:
Cleanup in functions which are getting values from node parameters

- Address issues in duration and integer parameters conversion
- Introduce `declare_integer_node_params(..)` and
`get_duration_from_node_param(..)` helper functions

Signed-off-by: Michael Orlov <michael.orlov@apex.ai>

MichaelOrlov commented:
@roncapat Ready for review and merge.
Although I've already done major cleanups, I have not gotten to the tests yet.  

Let's merge this portion of fixes first, then after Thanksgiving, I can get back to the tests.
Meanwhile please consider adding some basic tests in addition to what we already have to make sure that parameters correctly loading with minimum valuable configs. i.e. when many parameters are absent and shall be settled up to the default values. We can check default values directly and don't need to compare yaml files.
